src/Robo/Commands/Setup/BuildCommand.php

```diff
@@ -142,4 +142,50 @@ public function installAlias() {
     return $result;
   }
 
+  /**
+   * Checks the repo if there is modified files.
+   *
+   * @return string|null
+   *   List of modified files.
+   */
+  protected function checkCleanRepo() {
+    $clean = $this->taskExec("git status --porcelain")
+      ->dir($this->getConfigValue('repo.root'))
+      ->setVerbosityThreshold(VerbosityThresholdInterface::VERBOSITY_VERBOSE)
+      ->run();
+    return $clean;
+  }
+
+  /**
+   * Adds a project dependency and ensures configuration management is in sync.
+   *
+   * @param string $dependency
+   *   The composer dependency to be added to composer.json.
+   *
+   * @command add-dependency
+   */
+  public function addDependency($dependency) {
+    $modified = $this->checkCleanRepo();
+    if (!empty($modified)) {
+      $this->output()->writeln("<comment>Please note the following files need to be committed or reverted:</comment>");
+      $this->output()->writeln("<comment>$modified</comment>");
+    }
+    else {
+      $this->taskExec("composer update $dependency --with-dependencies")
```

Comment:
This will update a dependency, not add it. You need `composer require` to add it.